### PR TITLE
fix(issues) Display custom search title when query string matches

### PR DESCRIPTION
**Pull request by markstory**
In addition to matching saved searches by id, we should match by querystring so that if a user replicates a savedsearch they get a nice title.

Notably this fixes the 'Custom Search' title when users first load up the organization issue list which doesn't support default saved searches just yet.

Fixes APP-1052

### CODE REVIEW

File: tests/js/spec/views/stream/savedSearchSelector.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  let props;
+
+  let orgId;
+  let projectId;
+  let savedSearchList;
+  let onCreate = jest.fn();
+  let onSelect = jest.fn();
+
+  beforeEach(function() {
+    sandbox = sinon.sandbox.create();
```
Comment (billyvg) [inline tests/js/spec/views/stream/savedSearchSelector.spec.jsx]: This doesn't seem to be used

File: src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx [hunk 2]
```diff
@@ -222,12 +223,16 @@ const SavedSearchSelector = withApi(
...
-      if (!searchId) return t('Custom Search');
-      let results = this.props.savedSearchList.filter(search => {
-        return searchId === search.id;
-      });
-      return results.length ? results[0].name : t('Custom Search');
+      let {searchId, query, savedSearchList} = this.props;
+      let result;
+
+      if (searchId) {
+        result = savedSearchList.find(search => searchId === search.id);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx]: Is `search.id` always a string?

File: src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx [hunk 3]
```diff
@@ -222,12 +223,16 @@ const SavedSearchSelector = withApi(
...
+      } else {
+        result = savedSearchList.find(search => query === search.query);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx]: Does `query` come from the search input? If so, I know it sometimes does funky things with whitespace, do we want to trim it?

**Review by billyvg — APPROVED** [review]
Thanks for the tests!

File: src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx]: It is. One of our number strings.

File: src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx (hunk 3 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx]: It comes from the URL querystring at the top-level and snakes its way down to here. From testing on my saved searches I didn't need to trim. I can check again when this is deployed.

**Pull request merged**